drivers/input/tsc2007.c

```diff
@@ -1230,11 +1230,11 @@ int tsc2007_register(FAR struct i2c_master_s *dev,
   nxsem_init(&priv->devsem,  0, 1); /* Initialize device structure semaphore */
   nxsem_init(&priv->waitsem, 0, 0); /* Initialize pen event wait semaphore */
 
-  /* The event wait semaphore is used for signaling and, hence, should not
-   * have priority inheritance enabled.
+  /* The event dev semaphore is used for lock, hence,
+   * should have priority inheritance enabled.
    */
 
-  nxsem_set_protocol(&priv->waitsem, SEM_PRIO_NONE);
+  nxsem_set_protocol(&priv->devsem, SEM_PRIO_INHERIT);
```

Comment:
revert the change, let's do all change in https://github.com/apache/incubator-nuttx/pull/7344